### PR TITLE
Bastion node support for Private Subnets

### PR DESCRIPTION
**Pull request by sourabh912**

### CODE REVIEW

File: qds_sdk/cluster.py [hunk 1]
```diff
@@ -170,6 +170,9 @@ def _parse_create_update(cls, args, action, api_version):
         ec2_group.add_argument("--vpc-id",
                                dest="vpc_id",
                                help="vpc to create the cluster in",)
+        ec2_group.add_argument("--bastion-node-public-dns",
+                               dest="bastion_node_public_dns",
+                               help="public dns name of the bastian node. Required only if cluster is in private subnet",)
```
Comment (vogxn) [inline qds_sdk/cluster.py]: typo in `help` - `bastian` => `bastion`. 
Also to be specific: `Required only if the cluster is in the private subnet of a EC2-VPC`

File: qds_sdk/cluster.py [hunk 2]
```diff
@@ -1031,13 +1040,16 @@ def set_cluster_info(self, aws_access_key_id=None,
 
         `presto_custom_config`: Custom Presto configuration overrides.
 
+        `bastion_node_public_dns`: Public dns name of the bastian node. Required only if cluster is in private subnet.
```
Comment (vogxn) [inline qds_sdk/cluster.py]: Type exists here too `bastian` => `bastion`

File: tests/test_cluster.py [hunk 3]
```diff
@@ -1989,6 +2025,29 @@ def test_persistent_security_group(self):
                                                     }
                                                 }})
 
+    def test_bastion_node_public_dns(self):
```
Comment (vogxn) [inline tests/test_cluster.py]: could we also include a test for the `private-subnet-security-group` updated along with the `bastion_node_public_dns` on a cluster within the VPC with private subnet?

File: qds_sdk/cluster.py [hunk 4]
```diff
@@ -1031,13 +1040,16 @@ def set_cluster_info(self, aws_access_key_id=None,
...
+
         """
 
         self.disallow_cluster_termination = disallow_cluster_termination
         self.enable_ganglia_monitoring = enable_ganglia_monitoring
         self.node_bootstrap_file = node_bootstrap_file
         self.__set_node_configuration(master_instance_type, slave_instance_type, initial_nodes, max_nodes, slave_request_type, fallback_to_ondemand)
-        self.__set_ec2_settings(aws_access_key_id, aws_secret_access_key, aws_region, aws_availability_zone, vpc_id, subnet_id, role_instance_profile)
+        self.__set_ec2_settings(aws_access_key_id, aws_secret_access_key, aws_region, aws_availability_zone, vpc_id, subnet_id,
+                                bastion_node_public_dns, role_instance_profile)
```
Comment (harshshah87) [inline qds_sdk/cluster.py]: The order of arguments passed should be reversed. role_instance_profile followed by bastion_node_public_dns.

**Comment by harshshah87** [comment]
Changes looks good.

File: tests/test_cluster.py (hunk 3 above)
Comment (sourabh912) [inline tests/test_cluster.py]: Sorry I did not get you. Do you mean updating persistent-security-group ?

Comment (vogxn) [inline tests/test_cluster.py]: Exactly. Updating multiple parts of the private subnet requirements.

Comment (sourabh912) [inline tests/test_cluster.py]: @vogxn : I have added the test suggested by you. You can find it [here](https://github.com/sourabh912/qds-sdk-py/commit/3917d830ea6e7468212265a8f40dd1456ccde9fa).

**Comment by vogxn** [comment]
Looks good. +1 to merge

**Pull request merged**